BlueDB/src/main/java/org/bluedb/api/SegmentSize.java

```diff
@@ -0,0 +1,27 @@
+package org.bluedb.api;
+
+import org.bluedb.api.keys.BlueKey;
+import org.bluedb.api.keys.HashGroupedKey;
+import org.bluedb.api.keys.IntegerKey;
+import org.bluedb.api.keys.LongKey;
+import org.bluedb.api.keys.TimeKey;
+
+public class SegmentSize<T extends BlueKey> {
+
+	public static final SegmentSize<TimeKey> TIME_1_HOUR = new SegmentSize<>(3_600_000L);
+	public static final SegmentSize<TimeKey> TIME_1_DAY = new SegmentSize<>(24 * 3_600_000L);

```

Comment:
We should be able to offer more than just two time options. We should check with Mike to see what he wants. I could see him using 1 Week. I could also see us using 8 hours or something like that instead of going all the way to 1 day.

Reply:
I need to better understand the costs and benefits associated with different segment sizes. Right now my understanding is limited to:

- each time you read a file has a cost
- because the system has to deserialize everything in a given file there is cost there also

I don't know what the optimal balance is. Rather than teaching each user what that balance is, perhaps there could be some sort of SegmentBuilder that takes parameters like frequency of records, minimum query duration, maximum query duration, average query duration, or whatever else may go into choosing your optimal segment size.

Reply:
Interesting idea. I think for power users who really understand BlueDB details I'll want to give the client the option to select specific sizes. But it might be nice to also support some sort of builder that auto selects based on some information you give it. I'm happy to chat about the pros and cons of different segment sizes with you. We want to write up an article that helps explain this for those who want to know more.